src/routes/patient.ts

```diff
@@ -0,0 +1,270 @@
+            ownerName: req.body.ownerName,
+            ownerAddress: req.body.ownerAddress,
+            ownerPhone: req.body.ownerPhone
+        }, {
+        new: true
+    });
+
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+    res.send(patient);
+});
+
+router.delete('/:id', validateObjectId, async (req: Request, res: Response) => {
+    const patient = await Patient.findByIdAndRemove(req.params.id);
+
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+    res.send(patient);
+});
+
+router.get('/:id', validateObjectId, async (req: Request, res: Response) => {
+    const patient = await Patient.findById(req.params.id);
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+
+    res.send(patient);
+});
+
+router.post('/:id/appointments', validateObjectId, async (req: Request, res: Response) => {
+    const patient = await Patient.findById(req.params.id);
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+
+    const { error } = validateAppointment.validate(req.body);
+    if (error) return res.status(400).send(error.details[0].message);
+    const { startTime, endTime, description, feeStatus, amount } = req.body;
+
+    let appointment = new AppointmentDocument({
+        startTime,
+        endTime,
+        description,
+        feeStatus,
+        amount
+    });
+
+    patient.appointment.push(appointment);
+    await patient.save();
+
+    res.send(patient);
+});
+
+router.get('/:id/appointments', validateObjectId, async (req: Request, res: Response) => {
+    const patient = await Patient.findById(req.params.id);
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+
+    res.send(patient.appointment);
+});
+
+router.put('/:id/appointments/:appointmentId', validateObjectId, async (req: Request, res: Response) => {
+    const patient = await Patient.findById(req.params.id);
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+
+    const appointment = patient.appointment.find(app => app._id.toString() === req.params.appointmentId);
+    if (!appointment) return res.status(404).send('The appointment with the given ID was not found.');
+
+    const { error } = validateAppointment.validate(req.body);
+    if (error) return res.status(400).send(error.details[0].message);
+    
+    appointment.set(req.body);
+
+    await patient.save();
+
+    res.send(appointment);
+});
+
+router.delete('/:id/appointments/:appointmentId', validateObjectId, async (req: Request, res: Response) => {
+    const patient = await Patient.findById(req.params.id);
+    if (!patient) return res.status(404).send('The patient with the given ID was not found.');
+
+    const appointment = patient.appointment.find(app => app._id.toString() === req.params.appointmentId);
```

Comment:
create respository/patient.ts and move all DB operations to the file and same for appointment